singer-io/tap-asana
# TDL-15272: Fix bug in refresh token logic and TDL-15467: Fix backoff logic

# Description of change
[TDL-15272](https://jira.talendforge.org/browse/TDL-15272) : Fix bug in token refresh logic
- Added TokenExpiredError and NoAuthorizationError exception in backoff to refresh the access token.
- As per Dan's suggestion over the yield-backoff issue, added a decorator over two SDK functions that are making final API calls because tap is yielding data from SDK function call which returns generator so direct backoff is not working.

[TDL-15467](https://jira.talendforge.org/browse/TDL-15467) : Fix backoff logic not used for all streams
- Added error-handling decorator at every possible API call.

# Manual QA steps
- Unit test cases to verify the above-mentioned exceptions.
- Set sleep time for 1 hour and verified that tap can run more than 1 hour after resetting access token.
 - Unit test cases to verify error handling is working for the above-mentioned streams.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch
